packages/core/lib/debug/printer.js

```diff
@@ -233,14 +233,14 @@ class DebugPrinter {
     this.config.logger.log(DebugUtils.formatCurrentInstruction(instruction));
 
     // printout 3 upcoming instructions
-    for (let i = Math.min(currentIndex + 1, instructions.length);
-      i <= currentIndex + upcomingInstructions;
+    for (let i = currentIndex + 1;
+      i < Math.min(currentIndex + upcomingInstructions + 1, instructions.length);
       i++) {
       this.config.logger.log(DebugUtils.formatInstruction(instructions[i]));
     }
 
     // add an ellipse if there exist additional instructions after
-    if (currentIndex + upcomingInstructions < instructions.length) {
+    if (currentIndex + upcomingInstructions + 1 < instructions.length) {
```

Comment:
Oh, I didn't even notice that the logic was wrong on this!  I totally didn't catch that!  Thanks for fixing this.  That said, two comments:

1. First, nitpicky comment, but again, I feel like this would be clearer if it were written as `currentIndex + upcomingInstructions + 1 < instructions.length`.  Again, feel free to ignore this if you disagree.

2. More substantial comment: Now that I look, the logic for the beginning ellipse is also off by one, isn't it?  If you're adjusting this one, you should probably adjust that one as well.

Reply:
For
1. Could you elaborate it more?
2. good catch, the beginning ellipse is off by one.  I will change it.

Reply:
Oops!  Sorry, I wrote the wrong thing.  I've fixed my comment now.  Thanks for catching that.